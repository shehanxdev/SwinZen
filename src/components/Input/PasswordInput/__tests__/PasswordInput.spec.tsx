import { fireEvent, render } from '@testing-library/react-native';

import { PasswordField, PasswordFieldProps } from '../PasswordInput';

describe('PasswordField component', () => {
  const TEST_ID = 'PasswordFieldTestID';
  const mockOnChangeFunction = jest.fn();

  const getRenderedPasswordFieldComponent = (props?: Partial<PasswordFieldProps>) =>
    render(<PasswordField {...props} testID={TEST_ID} />);

  beforeEach(() => {
    jest.useFakeTimers();
  });

  it('should find the PasswordField component via testID', () => {
    const { getByTestId } = getRenderedPasswordFieldComponent();
    const passwordField = getByTestId(TEST_ID);
    expect(passwordField).toBeTruthy();
  });

  test(`should render correctly`, () => {
    const rendered = getRenderedPasswordFieldComponent();
    const renderedTree = rendered.toJSON();
    expect(renderedTree).toMatchSnapshot();
  });

  it('should call onChangeText event when typed', async () => {
    const dummyInput = 'cL123456@!';
    const { getByTestId } = getRenderedPasswordFieldComponent({ onChangeText: mockOnChangeFunction });

    const inputElement = getByTestId(TEST_ID);

    expect(inputElement).toBeTruthy();

    if (!inputElement) return;

    fireEvent.changeText(inputElement, dummyInput);

    expect(mockOnChangeFunction).toBeCalled();
  });

  //TODO::add show/hide password on EyeIcon click test once the text field refactoring PRs get merged.
});
