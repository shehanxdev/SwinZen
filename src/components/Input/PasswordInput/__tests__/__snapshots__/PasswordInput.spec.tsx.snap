// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordField component should render correctly 1`] = `
<View>
  <View
    style={
      {
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 6,
          },
          {
            "backgroundColor": "#16501D",
            "borderColor": "#ffffff00",
            "borderRadius": 10,
            "borderWidth": 1,
          },
        ]
      }
      testID="text-input-outline"
    />
    <View>
      <View
        style={
          [
            {
              "paddingBottom": 0,
            },
            {
              "minHeight": 48,
              "paddingTop": 8,
            },
          ]
        }
      >
        <TextInput
          editable={true}
          maxFontSizeMultiplier={1.5}
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholderTextColor="#ADB5BD"
          secureTextEntry={true}
          selectionColor="#6C757D"
          style={
            [
              {
                "flexGrow": 1,
                "margin": 0,
                "paddingHorizontal": 14,
                "zIndex": 1,
              },
              {
                "height": 48,
              },
              {
                "paddingBottom": 16,
                "paddingTop": 16,
              },
              {
                "color": "#DEE2E6",
                "fontFamily": "System",
                "fontSize": 16,
                "fontVariant": undefined,
                "fontWeight": undefined,
                "lineHeight": 18,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              false,
              {
                "marginRight": 44,
                "paddingRight": 0,
              },
            ]
          }
          testID="PasswordFieldTestID"
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          [
            {
              "alignItems": "center",
              "height": 24,
              "justifyContent": "center",
              "position": "absolute",
              "width": 24,
            },
            {
              "right": 12,
              "top": 15,
            },
          ]
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "disabled": false,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          focusable={true}
          hitSlop={
            {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              [
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "margin": 6,
                  "overflow": "hidden",
                },
                {
                  "borderRadius": 18,
                  "height": 36,
                  "width": 36,
                },
                undefined,
                {
                  "marginTop": "50%",
                },
              ],
            ]
          }
        >
          <View>
            <SvgMock
              fill="#ADB5BD"
              height={20}
              width={20}
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <Text
    style={
      [
        {
          "color": "#E9ECEF",
          "lineHeight": 24,
        },
        {
          "fontFamily": "SourceSansPro-Regular",
          "fontSize": 15,
          "fontWeight": "600",
          "textAlign": "auto",
        },
      ]
    }
  />
</View>
`;
